Bottom: 1b4ca72ecb26c142b94c1b9925c3e23fbecedb73
Top:    4cb0cf6e2a8a78cafc8b87fb488a8d9c7668aacd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-06-02 21:56:22 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index f1a0e22..2657f30 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1163,7 +1163,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
